junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/DisplayNameUtils.java

```diff
@@ -70,22 +70,33 @@ static String determineDisplayName(AnnotatedElement element, Supplier<String> di
 		return displayNameSupplier.get();
 	}
 
-	static String determineDisplayNameForMethod(Class<?> testClass, Method testMethod) {
-		DisplayNameGenerator generator = getDisplayNameGenerator(testClass);
+	static String determineDisplayNameForMethod(Class<?> testClass, Method testMethod,
+			Optional<Class<? extends DisplayNameGenerator>> defaultDisplayNameGeneratorClass) {
+		DisplayNameGenerator generator = getDisplayNameGenerator(testClass, defaultDisplayNameGeneratorClass);
 		return determineDisplayName(testMethod, () -> generator.generateDisplayNameForMethod(testClass, testMethod));
 
 	}
 
-	static Supplier<String> createDisplayNameSupplierForClass(Class<?> testClass) {
-		return () -> getDisplayNameGenerator(testClass).generateDisplayNameForClass(testClass);
+	static Supplier<String> createDisplayNameSupplierForClass(Class<?> testClass,
+			Optional<Class<? extends DisplayNameGenerator>> defaultDisplayNameGeneratorClass) {
+		return () -> getDisplayNameGenerator(testClass, defaultDisplayNameGeneratorClass).generateDisplayNameForClass(
+			testClass);
 	}
 
-	static Supplier<String> createDisplayNameSupplierForNestedClass(Class<?> testClass) {
-		return () -> getDisplayNameGenerator(testClass).generateDisplayNameForNestedClass(testClass);
+	static Supplier<String> createDisplayNameSupplierForNestedClass(Class<?> testClass,
+			Optional<Class<? extends DisplayNameGenerator>> defaultDisplayNameGeneratorClass) {
+		return () -> getDisplayNameGenerator(testClass,
+			defaultDisplayNameGeneratorClass).generateDisplayNameForNestedClass(testClass);
 	}
 
-	private static DisplayNameGenerator getDisplayNameGenerator(Class<?> testClass) {
+	private static DisplayNameGenerator getDisplayNameGenerator(Class<?> testClass,
+			Optional<Class<? extends DisplayNameGenerator>> defaultDisplayNameGeneratorClass) {
 		Preconditions.notNull(testClass, "Test class must not be null");
+
+		if (defaultDisplayNameGeneratorClass.isPresent()) {
+			return ReflectionUtils.newInstance(defaultDisplayNameGeneratorClass.get());
+		}
+
```

Comment:
:x: We should only use the default generator, if the `@DisplayNameGeneration` annotation is not present.

Reply:
👍